src/Raven.Server/ServerWide/Commands/RemoveNodeFromDatabaseCommand.cs

```diff
@@ -1,24 +1,31 @@
-﻿using Raven.Client.ServerWide;
+﻿using System.Collections.Generic;
+using Raven.Client.ServerWide;
 using Sparrow.Json.Parsing;
 
 namespace Raven.Server.ServerWide.Commands
 {
     public class RemoveNodeFromDatabaseCommand : UpdateDatabaseCommand
     {
         public string NodeTag;
+        public string DatabaseId;
 
         public RemoveNodeFromDatabaseCommand()
         {
         }
 
-        public RemoveNodeFromDatabaseCommand(string databaseName, string uniqueRequestId) : base(databaseName, uniqueRequestId)
+        public RemoveNodeFromDatabaseCommand(string databaseName, string databaseId, string uniqueRequestId) : base(databaseName, uniqueRequestId)
         {
+            DatabaseId = databaseId;
         }
 
         public override string UpdateDatabaseRecord(DatabaseRecord record, long etag)
         {
             record.Topology.RemoveFromTopology(NodeTag);
             record.DeletionInProgress?.Remove(NodeTag);
+            if (record.UnusedDatabaseIds == null)
+                record.UnusedDatabaseIds = new HashSet<string>();
+
+            record.UnusedDatabaseIds.Add(DatabaseId);
```

Comment:
DatabaseId will be null in 'old' commands, right?

Reply:
right! done